teeworlds7/packet.go

```diff
@@ -48,23 +54,25 @@ func (client *Client) processPacket(packet *protocol7.Packet) error {
 
 	if packet.Header.Flags.Control {
 		if len(packet.Messages) != 1 {
-			return fmt.Errorf("got control packet with %d messages.\n", len(packet.Messages))
+			return fmt.Errorf("got control packet with %d messages: expected only 1", len(packet.Messages))
```

Comment:
One or zero. There can be empty control messages if a resend is requested.

Reply:
does this mean that the logic was previously incorrect? because the check  above did explicitly check whether it's exactly one element.

Reply:
What's the best way to fix this :?

Reply:
The best way to fix this is to implement resends and correctly read the flags even for empty packets.
But that is out of scope for this pr.

Until then let's just not touch the error message.

Reply:
reverted message, removed newline & punctuation.